src/Property.php

```diff
@@ -122,6 +136,12 @@ protected function resolveTypeDefinition()
         $this->arrayTypes = str_replace('[]', '', $this->types);
 
         $this->setNullable(strpos($varDocComment, 'null') !== false);
+
+        if (in_array('immutable', $this->types) || in_array('Immutable', $this->types)) {
```

Comment:
This check for immutability must be set before 

```php
$this->hasTypeDeclaration = true;
```

The reason is PHP 7.4:

We want to allow the following:

```php
class MyDTO
{
    /** @var immutable */
    public string $foo;
}
```

In this case we want PHP's type system to handle type checking, so `$this->hasTypeDeclaration` has to be `false`. But we do want to keep the immutability support.

I think we want something like this:

```php
if (in_array('immutable', $this->types) || in_array('Immutable', $this->types)) {
    $this->isImmutable = true;

    unset(
        $this->types['immutable'],
        $this->types['Immutable']
    );
}

if (empty($this->types)) {
    return;
}

$this->hasTypeDeclaration = true;

// …
```

Also note that you can use one unset call to unset multiple variables.

Reply:
yeah thanks i forgot